src/main/java/org/jabref/gui/preferences/PreviewTabView.java

```diff
@@ -0,0 +1,328 @@
+            } catch (StringIndexOutOfBoundsException exception) {
+                LOGGER.warn("Parsing error.", exception);
+                dialogService.showErrorDialogAndWait(Localization.lang("Parsing error"), Localization.lang("Parsing error") + ": " + Localization.lang("illegal backslash expression"), exception);
+            }
+
+            if (selectedLayout instanceof TextBasedPreviewLayout) {
+                previewText = ((TextBasedPreviewLayout) selectedLayout).getText().replace("__NEWLINE__", "\n");
+            } else {
+                previewText = ((CitationStylePreviewLayout) selectedLayout).getSource();
+            }
+            editArea.replaceText(previewText);
+        }
+    }
+
+    public void jumpToSearchKey(ListView<PreviewLayout> list, KeyEvent keypressed) {
+        if (keypressed.getCharacter() == null) {
+            return;
+        }
+
+        if (System.currentTimeMillis() - lastKeyPressTime < 1000) {
+            listSearchTerm += keypressed.getCharacter().toLowerCase();
+        } else {
+            listSearchTerm = keypressed.getCharacter().toLowerCase();
+        }
+
+        lastKeyPressTime = System.currentTimeMillis();
+
+        list.getItems().stream().filter(item -> item.getName().toLowerCase().startsWith(listSearchTerm))
+                .findFirst().ifPresent(list::scrollTo);
+    }
+
+    public void dragOverAvailable(@SuppressWarnings("unused") PreviewLayout layout, DragEvent event) {
+        if ((event.getGestureSource() != layout) && event.getDragboard().hasContent(DragAndDropDataFormats.PREVIEWLAYOUT)) {
+            event.acceptTransferModes(TransferMode.MOVE);
+        }
+    }
+
+    public void dragOverChosen(DragEvent event) {
+        if (event.getDragboard().hasContent(DragAndDropDataFormats.PREVIEWLAYOUT)) {
+            event.acceptTransferModes(TransferMode.MOVE);
+        }
+    }
+
+    public void dragDetected(ListView<PreviewLayout> sourceListView, MouseEvent event) {
+        PreviewLayout layout = sourceListView.getSelectionModel().getSelectedItem();
+        if (layout != null) {
+            ClipboardContent content = new ClipboardContent();
+            Dragboard dragboard = sourceListView.startDragAndDrop(TransferMode.MOVE);
+            content.put(DragAndDropDataFormats.PREVIEWLAYOUT, layout.getName());
+            dragboard.setContent(content);
+            dragSourceList = sourceListView;
+        }
+        event.consume();
+    }
+
+    public void dragDropped(ListView<PreviewLayout> targetListView, DragEvent event) {
+        Dragboard dragboard = event.getDragboard();
+        boolean success = false;
+
+        if (dragboard.hasContent(DragAndDropDataFormats.PREVIEWLAYOUT)) {
+            PreviewLayout draggedLayout = viewModel.findLayoutByNameOrNull((String) dragboard.getContent(DragAndDropDataFormats.PREVIEWLAYOUT));
+            if (draggedLayout != null) {
+                if (dragSourceList != null) {
```

Comment:
If you look into the code from the LinkedFilesEditor you see how to get the position and add the element at the right position.
```java
  ObservableList<LinkedFileViewModel> items = listView.itemsProperty().get();

        if (dragboard.hasContent(DragAndDropDataFormats.LINKED_FILE)) {

            LinkedFile linkedFile = (LinkedFile) dragboard.getContent(DragAndDropDataFormats.LINKED_FILE);
            LinkedFileViewModel transferedItem = null;
            int draggedIdx = 0;
            for (int i = 0; i < items.size(); i++) {
                if (items.get(i).getFile().equals(linkedFile)) {
                    draggedIdx = i;
                    transferedItem = items.get(i);
                    break;
                }
            }
            int thisIdx = items.indexOf(originalItem);
            items.set(draggedIdx, originalItem);
            items.set(thisIdx, transferedItem);
            success = true;
        }
```

Reply:
Let's say you have a list with A,B,C: You move C to A. You now need to swap the index positions of the entries. I had to use this approach cause I put the serializable model class in the clipboard, so I need to find the corresponding viewModel positions in the listView first.